api.go

```diff
@@ -114,17 +111,54 @@ func (c *cron) Delete(ctx context.Context, name string) error {
 	return c.queue.Dequeue(jobKey)
 }
 
-// validateName validates the name of a job.
-func (c *cron) validateName(name string) error {
-	if len(name) == 0 {
-		return errors.New("job name cannot be empty")
+// DeletePrefixes deletes cron jobs with the given prefixes from the cron
+// instance.
+func (c *cron) DeletePrefixes(ctx context.Context, prefixes ...string) error {
+	select {
+	case <-c.readyCh:
+	case <-c.closeCh:
+		return errors.New("cron is closed")
+	case <-ctx.Done():
+		return ctx.Err()
 	}
 
-	for _, segment := range strings.Split(strings.ToLower(c.validateNameReplacer.Replace(name)), "||") {
-		if errs := validation.IsDNS1123Subdomain(segment); len(errs) > 0 {
-			return fmt.Errorf("job name is invalid %q: %s", name, strings.Join(errs, ", "))
+	for _, prefix := range prefixes {
+		if len(prefix) == 0 {
+			continue
+		}
+
+		if err := c.validateName(prefix); err != nil {
+			return err
+		}
+	}
+
+	var errs []error
+	removeFromCache := func(jobKey string) {
+		c.queueLock.Lock(jobKey)
+		defer c.queueLock.DeleteUnlock(jobKey)
+
+		if _, ok := c.queueCache.Load(jobKey); ok {
+			return
+		}
+
+		c.queueCache.Delete(jobKey)
+		errs = append(errs, c.queue.Dequeue(jobKey))
+	}
+
+	for _, prefix := range prefixes {
+		keyPrefix := c.key.JobKey(prefix)
+		c.log.V(3).Info("deleting jobs with prefix", "prefix", keyPrefix)
+
+		resp, err := c.client.Delete(ctx, keyPrefix, clientv3.WithPrefix(), clientv3.WithPrevKV())
```

Comment:
Don't we need to cleanup the counters here as well or am I missing that somewhere in the diff?